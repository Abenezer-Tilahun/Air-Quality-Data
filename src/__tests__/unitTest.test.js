import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import MockedHeader from '../__mock__/Header';
import MockedFooter from '../__mock__/Footer';

describe('Components snapshot tests', () => {
  const mockStore = configureStore();

  it('Header renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={mockStore({})}>
          <MockedHeader />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Footer renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={mockStore({})}>
          <MockedFooter />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
