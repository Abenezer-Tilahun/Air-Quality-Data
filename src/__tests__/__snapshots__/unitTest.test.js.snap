// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components snapshot tests Footer renders correctly 1`] = `
<footer
  className="grid place-content-center bg-pink-500 text-white shadow-lg"
>
  <h5>
    A Metrics WebApp Design By Abenezer-Tilahun(@2022
    <i
      aria-hidden="true"
      className="fa fa-copyright"
    />
    )
  </h5>
</footer>
`;

exports[`Components snapshot tests Header renders correctly 1`] = `
<div
  className="bg-pink-500 shadow-lg"
>
  <nav>
    <div
      className="d-flex justify-content-between align-items-center p-3 mb-2"
    >
      <h1
        className="header"
      >
        AirNow
      </h1>
    </div>
  </nav>
</div>
`;
